core/src/main/java/smile/classification/DecisionTree.java

```diff
@@ -744,13 +759,13 @@ private double impurity(int[] count, int n) {
 
         switch (rule) {
             case GINI:
-                impurity = 1.0;
+                double squared_sum = 0;
```

Comment:
@haifengl Did you review this modification?

Reply:
This is supposed to be equivalent - both old and new compute $1 - \sum_i c_i^2 / n^2$, but the old approach has a division per class, while the new has only a single division.

Reply:
hmm, right. I have little concern about overflow at `squared_sum += ` though.